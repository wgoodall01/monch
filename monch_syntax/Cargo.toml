[package]
name = "monch_syntax"
version = "0.1.0"
edition = "2021"

[dependencies]
pest = "2.1.3"
pest_derive = "2.1.0"
serde = { version = "1", features = ["derive"] }
thiserror = "1.0.30"

[dev-dependencies]
insta = { version = "1.8.0", features = ["glob"] }
