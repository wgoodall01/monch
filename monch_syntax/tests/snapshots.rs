use insta::{assert_yaml_snapshot, glob};
use monch_syntax::Parser;
use std::fs;

fn display_err<E: std::error::Error>(err: E) -> E {
    println!("Error:");
    println!("{}", err);
    println!("{:#?}", err);
    err
}

#[test]
fn snapshot_script() {
    glob!("fixtures/*.example.monch", |path| {
        let input = fs::read_to_string(path).unwrap();

        // Make a parser
        let parsed = Parser::new()
            .parse_script(&input)
            .map_err(display_err)
            .expect("Script failed to parse");

        assert_yaml_snapshot!(parsed);
    })
}
