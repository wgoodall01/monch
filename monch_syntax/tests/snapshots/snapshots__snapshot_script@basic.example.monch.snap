---
source: monch_syntax/tests/snapshots.rs
expression: parsed
input_file: monch_syntax/tests/fixtures/basic.example.monch

---
commands:
  - pipeline:
      - executable:
          Literal:
            value: echo
        arguments:
          - Literal:
              value: one
          - Literal:
              value: two
          - Literal:
              value: three
        stdin_redirect: ~
        stdout_redirect: ~
  - pipeline:
      - executable:
          Literal:
            value: echo
        arguments:
          - Literal:
              value: one
          - Literal:
              value: two
          - Literal:
              value: three
        stdin_redirect: ~
        stdout_redirect: ~
  - pipeline:
      - executable:
          Literal:
            value: echo
        arguments:
          - Literal:
              value: one
          - Literal:
              value: two
          - Literal:
              value: three
        stdin_redirect: ~
        stdout_redirect: ~
  - pipeline:
      - executable:
          Literal:
            value: echo
        arguments:
          - Literal:
              value: arg2
          - Literal:
              value: arg2
        stdin_redirect: ~
        stdout_redirect: ~
  - pipeline:
      - executable:
          Literal:
            value: ls
        arguments:
          - Literal:
              value: "-l"
          - Literal:
              value: "--inode"
        stdin_redirect: ~
        stdout_redirect: ~
  - pipeline:
      - executable:
          Literal:
            value: ls
        arguments:
          - Literal:
              value: "--quoting-style"
          - Literal:
              value: shell
        stdin_redirect: ~
        stdout_redirect: ~
  - pipeline:
      - executable:
          Literal:
            value: ls
        arguments: []
        stdin_redirect: ~
        stdout_redirect: ~
      - executable:
          Literal:
            value: xargs
        arguments:
          - Literal:
              value: cat
        stdin_redirect: ~
        stdout_redirect: ~
  - pipeline:
      - executable:
          Literal:
            value: ls
        arguments: []
        stdin_redirect: ~
        stdout_redirect:
          TruncateFile:
            file:
              Literal:
                value: file_listing.txt
  - pipeline:
      - executable:
          Literal:
            value: cat
        arguments: []
        stdin_redirect:
          File:
            file:
              Literal:
                value: file_listing.txt
        stdout_redirect: ~
  - pipeline:
      - executable:
          Literal:
            value: first quoted cmd
        arguments:
          - Literal:
              value: bare
          - Literal:
              value: doublequote
        stdin_redirect:
          File:
            file:
              Literal:
                value: input file
        stdout_redirect: ~
      - executable:
          Literal:
            value: second-cmd
        arguments:
          - Literal:
              value: arg
        stdin_redirect: ~
        stdout_redirect:
          TruncateFile:
            file:
              Literal:
                value: output-file

